src/components/Details/TxnDetailsTable/TxnDetailsTable.tsx

```diff
@@ -31,49 +35,67 @@ const ValidityIcon = ({ validity }: { validity: string }) => {
   )
 }
 
+const value = (value: string, action: string) => {
+  switch (action) {
+    case "Transfer":
+    case "Mint":
+      return `${value} CON`
+
+    default:
+      return value
+  }
+}
+
 const TxnDetailsTable = ({
   txhash,
   createdt,
   validation_code,
   chaincodename,
   payload_proposal_hash,
+  tx_to,
+  tx_from,
+  tx_action,
+  tx_value,
 }: Props) => {
   return (
     <div>
-      <div className={`${style.row} ${style.scrolly}`}>
-        <div className={style.key}>Transaction Hash:</div>
-        <div className={style.selectVal}>{txhash}</div>
-      </div>
-      <div className={style.row}>
-        <div className={style.key}>Timestamp:</div>
-        <div className={style.selectVal}>
-          <TimeStampCell time={createdt} timeStyle="round" />
-          <span>({new Date(createdt).toUTCString()})</span>
-        </div>
-      </div>
-      <div className={style.row}>
-        <div className={style.key}>Validity:</div>
-        <div className={style.val}>
-          <ValidityIcon validity={validation_code} />
-        </div>
-      </div>
-      <div className={style.row}>
-        <div className={style.key}>Contract:</div>
-        <div className={style.selectVal}>
-          <Link to={`/contracts/${chaincodename}`}>
+      <DetailsTableRow keyCell="Transaction Hash" value={txhash} select />
+      <DetailsTableRow keyCell="Timestamp">
+        <TimeStampCell time={createdt} timeStyle="round" />
+        <span>({new Date(createdt).toUTCString()})</span>
```

Comment:
we should probably make this more readable.

Reply:
Shall I turn that into its own function?

Reply:
Per our conversation, this is an acceptable format already